src/compile/mark/valueref.ts

```diff
@@ -149,7 +149,7 @@ export function text(textDef: Conditional<TextFieldDef<string>, ValueDef<any>>,
   // text
   if (textDef) {
     if (isFieldDef(textDef)) {
-      return formatSignalRef(textDef, textDef.format, 'datum', config);
+      return formatSignalRef(textDef, textDef.format, 'datum', config, textDef.formatType || 'number', 'text');
```

Comment:
I don't think it's always the text channel here.  (It could be a `tooltip` when ref.text is called via `mixins.text`)